src/error_handling/lexical.go

```diff
@@ -0,0 +1,75 @@
+package errorhandling
+
+import (
+	"log"
+	"strings"
+)
+
+type LexicalErrorType int
+
+const (
+	InvalidLiteral LexicalErrorType = iota
+	InvalidNumber
+	InvalidComment
+	InvalidWord
+)
+
+func isInvalidNumber(lexem string) bool {
+	containsQuotation := strings.Contains(lexem, "\"")
+	containsBrackets := strings.Contains(lexem, "{")
+	containsNumber := strings.ContainsAny(lexem, "0123456789")
+
+	if !containsBrackets && !containsQuotation && containsNumber {
+		return true
+	}
+	return false
+}
+
+func isInvalidLiteral(lexem string) bool {
+	numberOfQuotations := strings.Count(lexem, "\"")
+	indexOfQuotation := strings.Index(lexem, "\"")
+	if numberOfQuotations == 1 && indexOfQuotation == 0 {
+		return true
+	}
+	return false
+}
+
+func isInvalidComment(lexem string) bool {
+	numberOfBrackets := strings.Count(lexem, "{")
+	indexOfBracket := strings.Index(lexem, "{")
+	if numberOfBrackets == 1 && indexOfBracket == 0 {
+		return true
+	}
+	return false
+}
```

Comment:
For what I can understand, this  function just covers one single case: the case where that is no closing bracket for the comment. We still have the case in which there are N closing bracket, with only one opening bracket, for instance: "{any comment}}}"

Reply:
I think that implicit we cover this case, because we I run the code that you suggested I get the following:
```
Classe: Comentário, Lexema: {anycomment}, Tipo: NULO
2022/01/29 12:07:13 erro na linha 1 coluna 14, palavra } inexistente na linguagem
Classe: ERRO, Lexema: , Tipo: NULO
Classe: EOF, Lexema: , Tipo: NULO
```

Reply:
Considering that this specific example concerns the comment error, wouldn't the actual output be something in the lines of:

```
Classe: Comentário, Lexema: {anycomment}, Tipo: NULO
erro na linha 1 coluna 14, comentário } inválido
Classe: ERRO, Lexema: , Tipo: NULO
Classe: EOF, Lexema: , Tipo: NULO
```
Since, currently the error is being handled as a "word error" and not a "comment error".